### PR TITLE
Teambuilder: Implement Box system

### PR DESCRIPTION
**Pull request by MacChaeger**
http://prntscr.com/mbubum
Max box size is 24, the same as the maximum team size for custom game.

### CODE REVIEW

**Comment by MacChaeger** [comment]
I should probably explain how this works some more. "Boxes" is a special folder that cannot be renamed or deleted (at least, not with the buttons that normally do that). The folder will always be present, even if a user has no boxes in the folder. Boxes may be given formats to allow for easier setbuilding and organization.

**Comment by Zarel** [comment]
This is... pretty impressive but ugh new client!

**Comment by MacChaeger** [comment]
As far as I can tell, the teambuilder hasn’t been rewritten yet, or if it has, no code’s been published to #1156 regarding it. Sorry if this is more work for you, but I don’t know what I could have done short of rewriting the whole thing myself.

**Comment by MacChaeger** [comment]
Waited. I could have waited.

**Comment by MacChaeger** [comment]
now it looks like this https://cdn.discordapp.com/attachments/454460542988320769/773010975753764864/unknown.png

**Comment by MacChaeger** [comment]
Looks like I’m still forgetting to `npm test`. Can’t fix it now, though, I’m currently busy hanging out with two very small and very adorable kittens.

File: js/client-teambuilder.js [hunk 1]
```diff
@@ -447,7 +455,7 @@
 
 			buf += '</ul>';
 			if (atLeastOne) {
-				buf += '<p><button name="new" class="button"><i class="fa fa-plus-circle"></i> ' + newButtonText + '</button></p>';
+				buf += '<p><button name="new" value="team" class="button"><i class="fa fa-plus-circle"></i> ' + newTeamButtonText + '</button><button name="new" value="box" class="button"><i class="fa fa-plus-circle"></i> ' + newBoxButtonText + '</button></p>';
```
Comment (Zarel) [inline js/client-teambuilder.js:458]:
```suggestion
				buf += '<p><button name="new" value="team" class="button"><i class="fa fa-plus-circle"></i> ' + newTeamButtonText + '</button> <button name="new" value="box" class="button"><i class="fa fa-plus-circle"></i> ' + newBoxButtonText + '</button></p>';
```

File: js/client-teambuilder.js [hunk 2]
```diff
@@ -355,13 +355,19 @@
 				}
 			}
 
-			var newButtonText = "New Team";
-			if (filterFolder) newButtonText = "New Team in folder";
+			var newTeamButtonText = "New Team";
+			var newBoxButtonText = "New Box";
+			if (filterFolder) {
+				newTeamButtonText = "New Team in folder";
+				newBoxButtonText = "New Box in folder";
```
Comment (Zarel) [inline js/client-teambuilder.js:362]: One thing we could do is just display "New Box" and hope being next to the "New Team" button is clear enough.

File: js/client-teambuilder.js [hunk 3]
```diff
@@ -355,13 +355,19 @@
...
+			}
 			if (filterFormat && filterFormat !== 'gen8') {
-				newButtonText = "New " + BattleLog.escapeFormat(filterFormat) + " Team";
+				newTeamButtonText = "New " + BattleLog.escapeFormat(filterFormat) + " Team";
+				newBoxButtonText = "New " + BattleLog.escapeFormat(filterFormat) + " Box";
 			}
-			buf += '<p><button name="newTop" class="button big"><i class="fa fa-plus-circle"></i> ' + newButtonText + '</button> ' +
-					 '<input type="text" id="teamSearchBar" name="search" class="textbox searchinput" value="' + this.curSearchVal + '" placeholder="search teams"/></p>';
+			buf += '<p><button name="newTop" value="team" class="button big"><i class="fa fa-plus-circle"></i> ' + newTeamButtonText + '</button> ' +
+				'<button name="newTop" value="box" class="button big"><i class="fa fa-plus-circle"></i> ' + newBoxButtonText + '</button> ' +
```
Comment (Zarel) [inline js/client-teambuilder.js:369]: This one could stand to use another icon.

Comment (Zarel) [reply js/client-teambuilder.js:369]: Square plus is the boring way:

https://fontawesome.com/v4.7.0/icon/plus-square

Thinking like a shopping cart:

https://fontawesome.com/v4.7.0/icon/shopping-bag

https://fontawesome.com/v4.7.0/icon/shopping-basket

https://fontawesome.com/v4.7.0/icon/shopping-cart

Referencing "the PC" in pokemon games:

https://fontawesome.com/v4.7.0/icon/tablet

https://fontawesome.com/v4.7.0/icon/television

https://fontawesome.com/v4.7.0/icon/laptop

https://fontawesome.com/v4.7.0/icon/server

Abstract:

https://fontawesome.com/v4.7.0/icon/sitemap

https://fontawesome.com/v4.7.0/icon/object-group

https://fontawesome.com/v4.7.0/icon/window-restore

https://fontawesome.com/v4.7.0/icon/upload

https://fontawesome.com/v4.7.0/icon/tags

https://fontawesome.com/v4.7.0/icon/th

https://fontawesome.com/v4.7.0/icon/list-alt

Comment (Zarel) [reply js/client-teambuilder.js:369]: image

Front-runner imo

**Comment by Zarel** [comment]
This is getting to be pretty good!

File: js/client-teambuilder.js (hunk 3 above)
Comment (MacChaeger) [reply js/client-teambuilder.js:369]: What about this one? https://fontawesome.com/v4.7.0/icon/archive

Comment (Zarel) [reply js/client-teambuilder.js]: Oh I like that one!

**Comment by MacChaeger** [comment]
Bump?

**Comment by Zarel** [comment]
Yeah, this looks good now. Thanks a lot!

**Pull request merged**